lib/arca/report.rb

```diff
@@ -0,0 +1,96 @@
+module Arca
+  class Report
+
+    # Arca::Report takes an Arca::Model and compiles the analyzed callback data
+    # into a short overview report for the model.
+    def initialize(model)
+      @model = model
+    end
+
+    # Public: Arca::Model representing the ActiveRecord class being reported.
+    attr_reader :model
+
+    # Public: Hash representation of the object for interactive consoles.
+    def inspect
+      to_hash.to_s
+    end
+
+    # Public: Hash of compiled report data.
+    def to_hash
+      {
+        :model_class         => model_class,
+        :model_file_path     => Arca.relative_path(model_file_path),
+        :number_of_callbacks => number_of_callbacks,
+        :lines_between       => lines_between_callbacks,
+        :externals           => externals,
+        :conditionals        => conditionals,
+        :permutations        => permutations
+      }
+    end
+
+    # Public: String class name of model.
+    def model_class
```

Comment:
I'd probably go with model_name as it is the String name not the class instance.